cmake/kokkos_tribits.cmake

```diff
@@ -307,8 +308,24 @@ FUNCTION(KOKKOS_INTERNAL_ADD_LIBRARY LIBRARY_NAME)
   )
 
   IF (KOKKOS_ENABLE_CUDA)
-    SET(LIB_cuda "-lcuda")
-    TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PUBLIC cuda)
+    IF (KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+       SET(LIB_cuda "-lcuda -lcudart")
+       find_library( cuda_lib_ NAMES libcuda.so cuda HINTS ${KOKKOS_CUDA_DIR}/lib64 )
+       find_library( cudart_lib_ NAMES libcudart.so cudart HINTS ${KOKKOS_CUDA_DIR}/lib64 )
+       if (cuda_lib_)
+          TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PUBLIC ${cuda_lib_})
+       else()
+          TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PUBLIC cuda)
+       endif()
+       if (cudart_lib_)
+          TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PUBLIC ${cudart_lib_})
+       else()
+          TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PUBLIC cudart)
```

Comment:
I think it is wrong to have this fallback.  It will yield a link time error unless cudart happen to be in the link directories.

Reply:
I agree to all of these issues/comments.  It will take me a bit of work/experimentation to produce a solution that accommodates all of the possibilities.  I have opened issue #2195 to track this effort